src/gps/GPS.cpp

```diff
@@ -563,24 +589,25 @@ int32_t GPS::runOnce()
         }
         GPSInitFinished = true;
         if (config.position.gps_enabled == false) {
```

Comment:
Shouldn't this code be at the *beginning* of runOnce(). If GPS is disabled there shouldn't be any probing.

My tests: current behavior with GPS disabled:
DEBUG | ??:??:?? 4 [GPS] Probing for GPS at 9600 
INFO  | ??:??:?? 4 [GPS] Found a UBlox Module using baudrate 9600
DEBUG | ??:??:?? 4 [GPS] Module Info : 
DEBUG | ??:??:?? 4 [GPS] Soft version: 7.03 (45969)
DEBUG | ??:??:?? 5 [GPS] Hard version: 00040007
DEBUG | ??:??:?? 5 [GPS] Extensions:0
WARN  | ??:??:?? 6 [GPS] Unable to save GNSS module configuration.
DEBUG | ??:??:?? 6 [GPS] WANT GPS=0
INFO  | ??:??:?? 6 [GPS] Setting GPS power=0

After moving to the beginning:
DEBUG | ??:??:?? 4 [GPS] WANT GPS=0
INFO  | ??:??:?? 4 [GPS] Setting GPS power=0

Reply:
No, because GPS disabled does not mean there is no GPS. GPS disabled implies we want the GPS powered down, if it's present. And in some cases, to do a power down, we have to detect the GPS.

Reply:
This is really really bad :(

It means that *all* devices that do not have GPS run into this lengthy 3x GPS probing with all the delays in it. Please try to find a solution to resolve this....

Reply:
We already have a couple solutions in place to deal with this. First, if the GPS pins aren't defined, the code knows to skip it. Which is now default on just about every device without a built-in GPS. And second, it does it lazy, so the rest of the boot can complete anyways.

Reply:
Or need a triple switch for gps_enabled: enabled/disabled/ignored

Reply:
GPS is going to be active on boot by default. We don't send any special code to tell it to turn on. On devices with a PMU or GPS enable line, maybe default off, I suppose. But take the Station G1 for instance. It has a port intended to add an external GPS, but it's just power and serial lines. On boot, that GPS chip is on and looking for satellites. To disable it, we first have to know that it's there.

Reply:
> We already have a couple solutions in place to deal with this. First, if the GPS pins aren't defined, the code knows to skip it. Which is now default on just about every device without a built-in GPS. And second, it does it lazy, so the rest of the boot can complete anyways.

I need to try this one... maybe because the old firmware forcibly defined the GPS pins they are still stored in the filesystem. Need to try a flash erase and check if it resolves this issue. Thanks.

Reply:
With my heltec_v3 it worked!

DEBUG | ??:??:?? 1 Read RTC time as 1695057317
DEBUG | ??:??:?? 1 Running without GPS.

Reply:
That unit doesn't have a GPS by default? If so, we need to remove the GPS PIN define from its variant.ini. Let the user set it if a GPS is installed.

Reply:
Yes, all the TLora boards don't have GPS built-in.

Reply:
I'll queue a separate PR to fix that, later today.